hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java

```diff
@@ -83,6 +95,52 @@ private static String randomName() {
         return SCHEMA_NAME_FILES + "_" + UuidUtil.newUnsecureUuidString().replace('-', '_');
     }
 
+    @Override
+    public List<Permission> permissions(SqlCall call) {
+        // Try find operand by name first, most common option.
+        SqlNode astPath = findOperandByName(OPTION_PATH, call);
+        // If not successful -> try to find by position: SELECT * FROM TABLE(fn('path', 'glob'))
+        if (astPath == null) {
+            astPath = findOperandByPosition(0, call);
+        }
+        if (astPath == null) {
+            throw QueryException.error("Can't determine the file path, query is denied in secure environment");
+        }
+        if (astPath instanceof SqlLiteral) {
+            String path = astPath.toSqlString(PostgresqlSqlDialect.DEFAULT).getSql();
+            // Strip quotes
+            if (path.charAt(0) == '\'' && path.charAt(path.length() - 1) == '\'') {
+                path = path.substring(1, path.length() - 1);
+            }
+            return Collections.singletonList(ConnectorPermission.file(path, ActionConstants.ACTION_READ));
+        } else if (astPath instanceof SqlDynamicParam) {
+            // Note: it is pre-validation phase, we can't extract the path value from dynamic param yet.
+            //  Better to have more strict permissions check here.
+            return Collections.singletonList(ConnectorPermission.file("*", ActionConstants.ACTION_READ));
+        }
+        return Collections.emptyList();
+    }
+
+    private static SqlNode findOperandByName(String name, SqlCall call) {
+        for (int i = 0; i < call.operandCount(); i++) {
+            if (call.operand(i) instanceof SqlBasicCall) {
+                SqlCall assignment = call.operand(i);
+                SqlIdentifier id = assignment.operand(1);
+                if (name.equals(id.getSimple())) {
+                    return assignment.operand(0);
+                }
+            }
+        }
+        return null;
+    }
+
+    private static SqlNode findOperandByPosition(int pos, SqlCall call) {
+        if (call.operand(pos) instanceof SqlLiteral) {
```

Comment:
`Index 0 out of bounds for length 0` instead of nice error for queries like `SELECT * FROM TABLE(JSON_FLAT_FILE())`

Reply:
afeb397c9812fee94a0cd4239d2184f4660d997a